lib/timers.js

```diff
@@ -500,8 +500,30 @@ function rearm(timer) {
   }
 }
 
+function extractTimer(timer) {
+  if (Timeout.prototype.isPrototypeOf(timer) || typeof timer !== 'object' ||
+      timer === null || timer === undefined) {
```

Comment:
You could merge the last two as `timer == null`. It's a lite idiom used quite often in "core" to mean "something nully".
Also BTW `typeof undefined !== `object`, so the second clause covers it.